lib/utils/deploy.go

```diff
@@ -0,0 +1,144 @@
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func DeployChallenge(chall_name, team_name string, patched bool) error {
+
+	team_namespace := team_name + "-ns"
+
+	log.Println(team_namespace)
+	kubeclient, err := GetClient(g.KatanaConfig.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	deploymentsClient := kubeclient.AppsV1().Deployments(team_namespace)
+	imageName := chall_name + ":latest"
+	if patched {
+		/// Retrieve the existing deployment
+        existingDeployment, err := deploymentsClient.Get(context.TODO(), chall_name, metav1.GetOptions{})
+        if err != nil {
+            fmt.Println("Error in retrieving existing deployment.")
+            log.Println(err)
+            return err
+        }
+
+        existingDeployment.Spec.Template.Spec.Containers[0].Image = team_name + "/" + chall_name
+
+        _, err = deploymentsClient.Update(context.TODO(), existingDeployment, metav1.UpdateOptions{})
+        if err != nil {
+            fmt.Println("Error in updating deployment.")
+            log.Println(err)
+            return err
+        }
+
+        fmt.Println("Updated deployment with new image.")
+        return nil
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: team_namespace,
+			Name:      chall_name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": chall_name,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": chall_name,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:            chall_name + "-" + team_name,
+							Image:           imageName,
+							ImagePullPolicy: v1.PullPolicy("Never"),
+							Ports: []v1.ContainerPort{
+								{
+									Name:          "challenge-port",
+									Protocol:      v1.ProtocolTCP,
+									ContainerPort: 80,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	result, err := deploymentsClient.Create(context.TODO(), deployment, metav1.CreateOptions{})
+
+	if err != nil {
+		fmt.Println(" FAT GYA..SADGE :( ")
+		panic(err)
+	}
+
+	fmt.Printf("Created deployment %q.\n", result.GetObjectMeta().GetName()+" in namespace "+team_namespace)
+	return nil
+
+}
+
+func int32Ptr(i int32) *int32 { return &i }
+
+func GetClient(pathToCfg string) (*kubernetes.Clientset, error) {
+	if pathToCfg == "" {
+		pathToCfg = filepath.Join(
+			os.Getenv("HOME"), ".kube", "config",
+		)
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", pathToCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+func GetPods(lbls map[string]string, ns ...string) ([]v1.Pod, error) {
```

Comment:
This function shouldn't belong in this file, check if such function exists, I think it does.